examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/expressions/DigitalDetoxPlanner.kt

```diff
@@ -0,0 +1,79 @@
+package com.xebia.functional.xef.auto.expressions
+
+import com.xebia.functional.xef.auto.Description
+import com.xebia.functional.xef.auto.llm.openai.OpenAI
+import com.xebia.functional.xef.prompt.Prompt
+import com.xebia.functional.xef.prompt.lang.Infer
+import kotlinx.serialization.Serializable
+
+enum class DigitalActivity {
+  SocialMedia,
+  Work,
+  Entertainment,
+  Gaming,
+  News,
+  Browsing,
+  Infer
+}
+
+@Serializable data class DigitalHabits(val activity: DigitalActivity, val dailyHours: Float)
+
+@Serializable
+data class DetoxState(
+  val currentScreenTime: Float, // in hours
+  val primaryActivities: List<DigitalHabits>,
+  val detoxGoal: String // e.g., "Reduce Social Media consumption by 50%"
+)
+
+@Serializable data class GenerateDetoxPlan(val state: DetoxState)
+
+@Serializable
+data class DetoxRecommendationPrompt(
+  @Description(
+    [
+      "Craft a digital detox plan based on the user's current habits and desired goals.",
+      "Recommend actionable steps and alternative non-digital activities to aid the detox process."
+    ]
+  )
+  val dayByDayActions:
+    List<DetoxPlan>, // e.g., {"Day 1": ["1-hour nature walk", "Read a book for 30 minutes"]}
+  val summary: String
+)
+
+@Serializable data class DetoxPlan(val day: String, val actions: List<String>)
+
+suspend fun main() {
+  OpenAI.conversation {
+    val infer = Infer(OpenAI.FromEnvironment.DEFAULT_SERIALIZATION, conversation)
+    val detoxPlan: DetoxRecommendationPrompt =
+      infer(
```

Comment:
I don't understand why we have a `infer` block here if there is no infer to do. I don't know when it is needed or when it is not.